src/opm/input/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp

```diff
@@ -363,39 +359,22 @@ struct SummaryConfigContext {
 
     bool is_connection_completion(const std::string& keyword)
     {
-        if (keyword[0] != 'C')
-            return false;
-
-        if (keyword.back() != 'L')
-            return false;
-
-        if (is_udq(keyword))
-            return false;
-
-        if (keyword.size() != 5)
-            return false;
+        static const auto conn_compl_kw = std::regex {
+            R"(C[OGW][IP][RT]L)"
```

Comment:
Seems more restrictive than before. But it is hard for me to judge which one is more correct.

Reply:
> Seems more restrictive than before

It is more restrictive than before, especially the phase designation (`[OGW]`).  This expression matches the ones we actually support.

> But it is hard for me to judge which one is more correct.

I prefer this one.  The other needed some additional logic to ensure we didn't get false positives.